lib/deref.js

```diff
@@ -178,7 +178,8 @@ module.exports = {
         resolveTo, stack, _.cloneDeep(seenRef), stackLimit);
     }
     if (schema.$ref && _.isFunction(schema.$ref.split)) {
-      let refKey = schema.$ref;
+      let refKey = schema.$ref,
+        outerRequired = concreteUtils.getOuterRequireIfIsSupported(schema);
```

Comment:
i think required is not the only property that could be added into the schema, according to:


A couple of other things are coming in from JSON Schema into the OpenAPI schemas. First of all: arbitrary keys are allowed in a schema; OpenAPI has relaxed its constraints on which fields can appear where to accommodate JSON Schema format objects. Also it's possible to $ref to an object, then **put keys alongside it** which are considered in addition to what is defined in the component.

content:
  'application/json':
    schema:
      $ref: '#/components/schemas/style'
      required:
        - hue
None of these changes are huge, but having the ability to use JSON Schema within OpenAPI is brilliant for anyone wanting to use the two together so to have this commitment to supporting the ever-improving JSON Schema is excellent news.

It does not establish which are the keys but, apparently is not limited to the require one

Reply:
Done, now it supports outer properties of any type, and concatenate them when value is an array